benchalerts/clients.py

```diff
@@ -399,15 +400,74 @@ def __init__(self, adapter: Optional[HTTPAdapter] = None):
         if login_creds["email"] and login_creds["password"]:
             self.post("/login/", json=login_creds)
 
+    @dataclass
+    class RunComparison:
```

Comment:
Great questions; I didn't really think this through much. I was just thinking "okay this is a bespoke class that will only be used as a structured output of this one client method, so let's put it next to that method." But I agree that a class on a class is confusing and at the very least makes it a little tough for tests. Especially when we do #16 .

A better question might be: once we factor ConbenchClient into its own package, does it make sense to still have `get_comparison_to_baseline()` as a method on it? It sort of seems like a `benchalerts`-specific function that wouldn't have much use elsewhere. So maybe we can do something like this:

- in this PR, move `get_comparison_to_baseline()` and `RunComparison` into their own file as a function taking a `ConbenchClient`
- in #16 , move `BaseClient` and `ConbenchClient` into their own package (without `get_comparison_to_baseline()` and `RunComparison`)
- keep only `GitHubClient` in this file and use this file for only clients talking to alerting services (until it gets too big again and we can turn it into a submodule)

Reply:
Yep, I could do that too. It'll be pretty much the same code except using `self` instead of `client` and everything will be indented more. Maybe I'll do the "file of functions" instead of "class of methods" just for the indentation.